components/Pages/Trade/Liquidity/Claim.tsx

```diff
@@ -82,25 +82,21 @@ const Claim = ({ poolId }: Props) => {
       <AvailableRewards totalValue={totalValue} />
 
       <ClaimTable tokens={rewards} />
-
       <SubmitButton
         label={buttonLabel}
         isConnected={true}
         txStep={TxStep.Ready}
-        isDisabled={isWalletConnected && !isClaimable && allSnapshotsTaken}
+        isDisabled={!isWalletConnected && !isClaimable && allSnapshotsTaken}
         isLoading={claim.isLoading}
         onClick={() => {
-          if (!isWalletConnected) {
-            openView()
```

Comment:
if no wallet connected, we want to display Connect Wallet on every primary button and open the wallet modal on click